src/ray/common/asio/instrumented_io_context.h

```diff
@@ -76,6 +76,52 @@ class instrumented_io_context : public boost::asio::io_context {
   void post(std::function<void()> handler, const std::string name = "UNKNOWN")
       LOCKS_EXCLUDED(mutex_);
 
+  /// Increments the current count and returns the enqueueing time for this handler.
+  ///
+  /// \param name A human-readable name to which collected stats will be associated.
+  /// \return The time at which this handler was enqueued and the handler stats.
+  std::pair<int64_t, std::shared_ptr<GuardedHandlerStats>> Enqueue(
```

Comment:
This version of `Enqueue` also returns the handler stats struct to be passed to `Instrument` in order to obviate the need for multiple handler stats table lookups (and therefore reducing table lock contention).